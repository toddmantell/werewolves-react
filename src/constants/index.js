export const CHARACTERS = {
	ANGEL: 'angel',
	BIG_BAD_WOLF: 'big bad wolf',
	COLLECTOR: 'collector',
	CUPID: 'cupid',
	DIRE_WOLF: 'dire wolf',
	DEFENDER: 'defender',
	DRACULA: 'dracula',
	GYPSY: 'gypsy',
	KNIGHT: 'knight',
	PSYCHIC_SISTERS: 'psychic sisters',
	THREE_BROTHERS: 'three brothers',
	VIGILANTE: 'vigilante',
	WEREWOLF: 'werewolf',
	WHITE_WOLF: 'white wolf',
	WILD_CHILD: 'wildchild',
	WITCH: 'witch',
	VILLAGER: 'villager'
}

export const NIGHTS = { 
	FIRST_NIGHT_ONLY: 0,
	EVERY_NIGHT: 1,
	AFTER_FIRST_EVERY_NIGHT: 2,
	EVERY_OTHER_NIGHT: 3,
	EVERY_THIRD_NIGHT: 4,
	DEPENDENT: 20,
	KNIGHT: 50
}
