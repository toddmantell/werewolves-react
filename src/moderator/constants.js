export const FIRST_NIGHT_ONLY = 0;
export const EVERY_NIGHT = 1;
export const AFTER_FIRST_EVERY_NIGHT = 2;
export const EVERY_OTHER_NIGHT = 3;
export const EVERY_THIRD_NIGHT = 4;
export const DEPENDENT = 20;
export const KNIGHT = 50;