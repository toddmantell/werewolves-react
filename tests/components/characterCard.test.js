import React from 'react';
import CharacterCard from '../../src/components/Character/characterCard';
import * as constants from '../../src/constantsAndUtilities/constants';
import toJson from 'enzyme-to-json';

describe('characterCard Tests', () => {
	const cardProps = { characterName: 'Bob', night: constants.FIRST_NIGHT_ONLY, characterImage: 'somepath/path', description: "He's a character" };

	it('should render 4 divs when mounted', () => {
		const fullyRenderedCard = mount(<CharacterCard {...cardProps} />);
		
		expect(fullyRenderedCard.find('div').length).toBe(4);
	});

	it('should render 2 divs and 1 img when shallow rendered', () => {
		const shallowRenderedCard = shallow(<CharacterCard {...cardProps} />);

		expect(shallowRenderedCard.find('div').length).toBe(2);
		expect(shallowRenderedCard.find('img').length).toBe(1);
	});
	
	it('should match the snapshot', () => {
		const renderedCard = render(<CharacterCard {...cardProps} />);

		expect(toJson(renderedCard)).toMatchSnapshot();
	});
});