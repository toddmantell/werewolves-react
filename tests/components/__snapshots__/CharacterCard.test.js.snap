// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`characterCard Tests should match the snapshot 1`] = `
<div
  class="sc-bdVaJa juTEnT"
>
  <div
    class="sc-bwzfXH ebHYCu"
  >
    Bob 
    <img
      src="somepath/path"
    />
    <div>
      Night: 
    </div>
    <div>
      He's a character
    </div>
  </div>
</div>
`;
